Aztec/Classes/TextKit/TextStorage.swift

```diff
@@ -493,6 +497,31 @@ open class TextStorage: NSTextStorage {
         }
     }
 
+    /// Processes differences in link styles, and applies them to the DOM in the specified
+    /// range.
+    ///
+    /// - Parameters:
+    ///     - range: the range in the DOM where the differences must be applied.
+    ///     - originalStyle: the original link URL object if any.
+    ///     - newStyle: the new link URL object if any.
+    ///
+    private func processLinkDifferences(in range: NSRange, betweenOriginal originalStyle: URL?, andNew newStyle: URL?) {
+
+        let addStyle = originalStyle == nil && newStyle != nil
+        let removeStyle = originalStyle != nil && newStyle == nil
+
+        if addStyle {
+            var attributes: [Libxml2.Attribute] = []
+            if let url = newStyle {
+                attributes.append(Libxml2.StringAttribute(name:"href", value: url.absoluteString))
+            }
+            dom.apply(element: .a, spanning: range, attributes: attributes)
```

Comment:
`TextStorage` was designed so it doesn't really need to know about HTML elements.  The mapping between styles and DOM elements is withing `DOMString`'s responsibility.

Can we replace this call with a more generic:

```swift
dom.applyLink(url: newUrl, spanning: range)
```